### DISCUSSION TITLE
Symfony and its uses

### DISCUSSION THREAD

**Discussion by 737simpilot** [CIDRAM/CIDRAM#311] [General]
I was looking at what Symfony did and discovered it's used in lot of websites and what have you including CIDRAM. I knew it was used in phpBB as well, but just didn't really know what it was until now. Well, I still don't "exactly" know what it is other than a base or something to help do things in PHP without having to code all of that in PHP I guess.  It appears to me like Electron where is in Electron you can create cross platform programs (think Windows, Linux, etc) with standard web code like; JavaScript, HTML, and CSS and Electron is the framework to make it all happen. https://www.electronjs.org/apps

Anyway, I was looking through [their projects](https://symfony.com/projects) and saw an interesting CMS software called [Grav](https://getgrav.org/). This is appealing to me because unlike WordPress, it doesn't have a boat ton of CVEs which are a _whopping_ [18 as of 2018](https://cve.mitre.org/cgi-bin/cvekey.cgi?keyword=grav). Compared to WordPress' _bazillion_ . Also, it's interesting in that like CIDRAM (unless you use PDO), Grav is a flat-file CMS! No database to mess around with and I guess the speed is great. I was going to create a blog and the Grav CMS may just fit the bill. Plus, Grav uses markdown just like here in Github so that could help me learn Markdown better. I'm used to BBcode and have hand typed it for over a decade on various forums running mostly vBulletin. My own website uses phpBB and I have a wealth of BBcode possibilities. Probably has more or just as many features as Xenforo which I don't much care about. Plus, phpBB is free.

Anyway, here's a pic.

![Just a drunk](https://i.postimg.cc/52Rj7HfC/ezgif-com-gif-maker-17.gif)

**Comment by Maikuolan** [comment]
The team behind Symfony have done some amazing work, and there's a lot of other amazing work out there that utilises work from Symfony in some form or another, too. Definitely a team worth following. :-)

In the case of CIDRAM, the main Symfony framework created and maintained by Symfony isn't used in entirety (not because of any kind of problem, but rather because doing so wouldn't be necessary whatsoever for a package like CIDRAM), but rather, just the polyfill they created for Idn. The reason for CIDRAM using Symfony's Idn polyfill is because CIDRAM's referrer spam module requires some of the functionality normally provided natively by PHP's Intl extension, but given that many PHP installations don't include the Intl extension by default, and it's not reasonable to suddenly demand the users of said PHP installations to install Intl just for the sake of being able to use a single module, an alternative, fallback, or polyfill was needed to cover cases where Intl isn't available, and so, I turned to Symfony's Idn polyfill for that. It does the job well, provides what's needed, and given its existence, permissible licensing, and the quality of code, saves me the need to bother writing something similar myself.

CIDRAM's requirements aside though, when starting a new project of a nature requiring such a framework, Symfony is definitely worth a glance at the very least (although it, as well as most other frameworks like it, tend to be a little weighty in terms of what they provide, the amount of code, files, learning curve, etc.. not necessarily bad, but a natural consequence to trying to build a framework as capable as that all the same.. so I tend to avoid using any of them unless it's necessary, seeing as I've always *tried*, at least, to adhere to not making things more complicated than necessary).

I've never used Grav before, but it definitely looks interesting. :-)
